x-pack/platform/plugins/shared/fleet/server/tasks/sync_integrations/sync_integrations_on_remote.ts

```
@@ -180,3 +220,224 @@ export const syncIntegrationsOnRemote = async (
+          result[ccrCustomName] = {
+            ...ccrCustomAsset,
+            sync_status: 'completed' as SyncStatus.COMPLETED,
+          } as RemoteSyncedCustomAssetsStatus;
+        } else {
+          result[ccrCustomName] = {
+            ...ccrCustomAsset,
+            sync_status: 'failed' as SyncStatus.FAILED,
+          } as RemoteSyncedCustomAssetsStatus;
+        }
+      }
+      if (ccrCustomAsset.type === 'component_template') {
+        const installedAsset = componentTemplatesByName[ccrCustomAsset?.name];
+
+        if (isEqual(installedAsset, ccrCustomAsset?.template)) {
+          result[ccrCustomName] = {
+            ...ccrCustomAsset,
+            sync_status: 'completed' as SyncStatus.COMPLETED,
+          } as RemoteSyncedCustomAssetsStatus;
+        } else {
+          result[ccrCustomName] = {
+            ...ccrCustomAsset,
+            sync_status: 'failed' as SyncStatus.FAILED,
+          } as RemoteSyncedCustomAssetsStatus;
+        }
+      }
+    });
+    return result;
+  } catch (err) {
+    logger.error('error', err.message);
+    throw err;
+  }
+};
+
+export const getRemoteSyncedIntegrationsStatus = async (
+  esClient: ElasticsearchClient,
+  soClient: SavedObjectsClientContract
+): Promise<GetRemoteSyncedIntegrationsStatusResponse> => {
+  const { enableSyncIntegrationsOnRemote } = appContextService.getExperimentalFeatures();
+  const logger = appContextService.getLogger();
+
+  if (!enableSyncIntegrationsOnRemote) {
+    return { integrations: [] };
+  }
+
+  try {
+    const followerIndexRes = await getFollowerIndexInfo(esClient, logger);
+    if (followerIndexRes?.error || !followerIndexRes?.info) {
+      return { error: followerIndexRes?.error, integrations: [] };
+    }
+    const res = await fetchAndCompareSyncedIntegrations(
+      esClient,
+      soClient,
+      followerIndexRes.info.follower_index,
+      logger
+    );
+    return res;
+  } catch (err) {
+    throw err;
```

should we return the error in the response?